nflow-engine/src/main/java/io/nflow/engine/workflow/curated/MaintenanceWorkflow.java

```diff
@@ -0,0 +1,37 @@
+package io.nflow.engine.workflow.curated;
+
+import io.nflow.engine.service.MaintenanceConfiguration;
+import io.nflow.engine.service.MaintenanceService;
+import io.nflow.engine.workflow.definition.NextAction;
+import io.nflow.engine.workflow.definition.StateExecution;
+import io.nflow.engine.workflow.definition.StateVar;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+
+import static io.nflow.engine.workflow.curated.CronWorkflow.State.schedule;
+import static io.nflow.engine.workflow.definition.NextAction.moveToState;
+
+/**
+ * Workflow that does nflow maintenance periodically.
+ */
+@Component
+public class MaintenanceWorkflow extends CronWorkflow {
+  public static final String MAINTENANCE_WORKFLOW_TYPE = "maintenance";
```

Comment:
Use a bit more specific type to avoid possible collisions.
```suggestion
  public static final String MAINTENANCE_WORKFLOW_TYPE = "nFlowMaintenance";
```
Could be "cleanupWorkflowInstances" also.